cl_manycore/libraries/bsg_manycore.cpp

```diff
@@ -802,7 +802,8 @@ int hb_mc_manycore_packet_tx(hb_mc_manycore_t *mc,
 		}
 	}
 
-	do { //	wait until transmit is complete: continuously write a packet length until done
+	int trials = 10;
```

Comment:
Is there a reason why the magic number here is 10?

Reply:
There is no special reason for that. It is a maximum number of write tails to make the regression test fail.

Reply:
So the purpose of this added while loop is to satisfy a regression test?

Reply:
Yes, only for the test_manycore_credits.c.
Where the credits are used up, then the host should not write anymore.

Reply:
But the host shouldn't care about the number of credits - the hardware should.

@max-ruttenberg Is this related to #177 ?

Reply:
@leonardxiang The host should check if the FIFO is full

Reply:
In this case, the Tx FIFO not full, but the host's credits is used up, so the host interface will not accept the read request sending from the host.
Do you mean that the host should check the credits is not 0 before sending the read request?

Reply:
But the purpose of this Issue/PR is to make the AXI-Manycore interface manages credits. When there are no credits available, packets should not be dequeued from the hardware FIFO. 

The host should only check if the Hardware FIFO is full.

Reply:
The host should never have to check that the credits are non-zero.

Reply:
Agreed.

Reply:
So the purpose of this check is merely to return a fail to the regression test after a certain amount of time?

Reply:
Yes.

Reply:
And if these lines are not present, what happens in the regression test?

Reply:
It will hang forever

Reply:
OK. It will hang forever because the packet was never sent. 

@max-ruttenberg, same question, #177 would address this correct?

Reply:
Yes it would. There’s a timeout argument already, we just need to use it.

Reply:
So should we address it in this PR, or in another?

Reply:
The RX part should also be fixed.

https://github.com/bespoke-silicon-group/bsg_f1/blob/0be751214ce3375a22fca8b228b3873a5e17301b/cl_manycore/libraries/bsg_manycore.cpp#L867-L876

I think they should be addressed in another PR.

Reply:
OK, can you update your PR to remove the while loop?

The test passes without this while loop, right?

Reply:
I meant, remove the trials while loop. We still need to check for tx_complete

Reply:
Sorry. I'll try to be clearer on this. 

The while loop was there previously right? Can you return this file to its unmodified state? I don't believe any software needs to change for this PR